code-challenges.js

```diff
@@ -26,3 +115,47 @@ var testingArray2 = [5, "Hola", 43, -34, "greetings", true]
 var testingArray3 = [3, 7, "hi", 10, 3, "hello", 4, "hi"]
 var testingArray4 = [7, "hi", 3, 1, "hi", 4, "hello", 4, 7]
 // Expected output: [ 3, 7, "hi", 10, "hello", 4, 1 ]
+
+
+// function that takes in two arrays
+const noDups = (arr1, arr2) => {
+  // spread both arrays into one
+  let arr = [...arr1, ...arr2];
+  // use the Set object to remove duplicates and convert it back into an array
+  return arr = [...new Set(arr)];
+}
+
+console.log(noDups(testingArray3, testingArray4));
+
+
+const noDupsV2 = (arr1, arr2) => {
+  let temp = [];
+  // concatanate both arrays into one
+  let arr = arr1.concat(arr2);
+  // loop through array
+  // if element within array is not present in the temp array
+  // push that element into the temp array
+  for (let i = 0; i < arr.length; i++) {
+    if (temp.indexOf(arr[i]) === -1) {
+      temp.push(arr[i]);
+    }
+  }
+
+  return temp;
+}
+
+// VERSION 3.0
+console.log(noDupsV2(testingArray3, testingArray4))
+
+const noDupsV3 = (arr1, arr2) => {
+  let temp = [];
+  let arr = arr1.concat(arr2);
+  arr.map(el => {
+    if (temp.includes(el) === false) {
+      temp.push(el);
+    }
+  })
+  return temp;
+}
+
+console.log(noDupsV3(testingArray3, testingArray4))
```

Comment:
good job putting all the f=different approaches! it shows that you've really thought it through :)